.github/workflows/build.yaml

```
@@ -86,28 +119,80 @@ jobs:
+          DB_IMAGE_NAME=$(echo "$DB_BUILD_CONFIG" | yq eval -r '.["image-name"]' -)
+          DB_WORK_DIR=$(echo "$DB_BUILD_CONFIG" | yq eval -r '.["work-dir"]' -)
+          
+          # Step 8: Set Default Dockerfile if service dockerfile config is missing
+          if [ -z "$SERVICE_DOCKERFILE" ]; then
             echo "No dockerfile entry found for \"$SERVICE\" in build-config.yml; using default"
-            DF="$DEFAULT_DOCKERFILE"
-          else
-            echo "Found dockerfile \"$DF\" for service \"$SERVICE\"."
+            SERVICE_DOCKERFILE="$SERVICE_WORK_DIR/$DEFAULT_DOCKERFILE"
+          fi
+          
```

YAMLlint (1.37.1)

[error] trailing spaces

(trailing-spaces)